### PR TITLE
Added support for track 2 to Mercury Payments

### PR DESCRIPTION
**Pull request by ryanbalsdon**
Review please @bizla @andrewpaliga @datwelk

This feels a bit hacky but it's by far the simplest way to go. Option 2 is to pull in the Magnet gem, which I just added a track1/2 field to for this purpose.

### CODE REVIEW

File: lib/active_merchant/billing/gateways/mercury.rb [hunk 1]
```diff
@@ -195,7 +195,13 @@ def add_amount(xml, money, options = {})
...
         xml.tag! 'Account' do
           if credit_card.track_data.present?
-            xml.tag! 'Track1', credit_card.track_data
+            if credit_card.track_data[0] == '%' # track 1 start sentinel (STX)
+              xml.tag! 'Track1', credit_card.track_data
+            elsif credit_card.track_data[0] == ';' # track 2 start sentinel (STX)
+              xml.tag! 'Track2', credit_card.track_data
+            else
+              xml.tag! 'Track1', credit_card.track_data
+            end
```
Comment (datwelk) [inline lib/active_merchant/billing/gateways/mercury.rb]:
```
if credit_card.track_data[0] == ';'
  .. # track 2
else
 .. # track 1
```

File: test/unit/gateways/mercury_test.rb [hunk 2]
```diff
@@ -63,12 +63,39 @@ def test_successful_refund
...
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |endpoint, data, headers|
+      assert_match(/<Track2>#{Regexp.escape(track_data)}<\/Track2>/, data)
+    end.respond_with(successful_purchase_response)
+
+    assert_instance_of Response, response
+    assert_success response
+  end
+
+  def test_cardpresent_with_bad_track_data_uses_track1_endpoint
```
Comment (datwelk) [inline test/unit/gateways/mercury_test.rb]: `test_bad_track_data_interpreted_as_track_1_data`?

File: test/unit/gateways/mercury_test.rb [hunk 3]
```diff
@@ -63,12 +63,39 @@ def test_successful_refund
     assert refund.test?
   end
 
-  def test_add_swipe_data_with_credit_card
-    @credit_card.track_data = "data"
+  def test_cardpresent_with_track_1_data
```
Comment (datwelk) [inline test/unit/gateways/mercury_test.rb]: nitpick: card present are 2 words

**Comment by andrewpaliga** [comment]
:+1:

File: lib/active_merchant/billing/gateways/mercury.rb (hunk 1 above)
Comment (bizla) [inline lib/active_merchant/billing/gateways/mercury.rb]: +1

File: test/unit/gateways/mercury_test.rb (hunk 2 above)
Comment (bizla) [inline test/unit/gateways/mercury_test.rb]: Yep, bit of a misnomer on the test name. Something like `test_bad_track_data_in_track_1` should suffice.

**Pull request closed without merging**